p2p/sync.go

```diff
@@ -409,26 +409,25 @@ func (s *syncService) genHeadersAndSigs(ctx context.Context, blockNumber uint64)
 	go func() {
 		defer close(headersAndSigCh)
 
-		headersIt(func(res *spec.BlockHeadersResponse) bool {
+	loop:
+		for res := range headersIt {
 			headerAndSig := specBlockHeaderAndSigs{}
 			switch v := res.HeaderMessage.(type) {
 			case *spec.BlockHeadersResponse_Header:
 				headerAndSig.header = v.Header
 			case *spec.BlockHeadersResponse_Fin:
-				return false
+				break loop
 			default:
 				s.log.Warnw("Unexpected HeaderMessage from getBlockHeaders", "v", v)
-				return false
+				break loop
 			}
 
 			select {
 			case <-ctx.Done():
-				return false
+				break loop
```

Comment:
label was not necessary, but let's keep it as is

Reply:
[done](https://github.com/NethermindEth/juno/pull/1954/commits/3e10bbbb4d92e126b5da30f74ab415f41527032e)